src/main/webapp/mentors.js

```diff
@@ -122,3 +122,32 @@ function createProjectsDiv(projectIDs) {
   }
   return projectsDiv;
 }
+
+function submitForm() {
+  const inputUrl = document.getElementById("inputRepo").value;
+  const encodedBody = new URLSearchParams();
+  encodedBody.append('inputRepo', inputUrl);
+  let url = new URL(
+      '/mentors', window.location.protocol + '//' + window.location.hostname);
+  const fetchRequest = fetch((url), {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/x-www-form-urlencoded',
+    },
+    body: encodedBody,
+  });
+  
+  fetchRequest.then((response) => response.json()).then((response) => {
+    if (response == 'success') {
```

Comment:
This definitely works, although it makes me feel like you're sort of reinventing the function of HTTP status codes. For example, if there is some sort of validation error in a user's input, you would want to respond with a 400 error. Then you can use `if (response.ok)` in a very similar way to how you're using `if (response == 'success')` here.

Reply:
I understand that this is a product of a hybridized approach to error handling, but I'm wondering why you wouldn't want to just use the JSON-based errors for everything? Here's what it could look like if you did:
```javascript
fetchRequest.then((response) => response.json()).then((response) => {
    window.location.href = '/dashboard.html';
}).catch((errorResponse) => {
   console.error(errorResponse);
   const errorContainer = document.getElementById('error-message-container');
   errorContainer.innerText = errorResponse.userMessage;
});
```
Where all errors would be handled by the same catch block instead of in two different places. All of the other functionality stays the same.

Reply:
I did something kind of similar! Still used the catch block, but have a seperate section for improper input handling (bc I don't want it to trigger an error on the console or an alert)